src/crawlers/puppeteer_crawler.js

```diff
@@ -1,557 +1,41 @@
- *         // 'page' is an instance of Puppeteer.Page with page.goto(request.url) already called
- *         // 'request' is an instance of Request class with information about the page to load
- *         await Apify.pushData({
- *             title: await page.title(),
- *             url: request.url,
- *             succeeded: true,
- *         })
- *     },
- *     handleFailedRequestFunction: async ({ request }) => {
- *         // This function is called when the crawling of a request failed too many times
- *         await Apify.pushData({
- *             url: request.url,
- *             succeeded: false,
- *             errors: request.errorMessages,
- *         })
- *     },
- * });
- *
- * await crawler.run();
- * ```
- * @property {AutoscaledPool} autoscaledPool
- *  A reference to the underlying {@link AutoscaledPool} class that manages the concurrency of the crawler.
- *  Note that this property is only initialized after calling the {@link PuppeteerCrawler#run} function.
- *  You can use it to change the concurrency settings on the fly,
- *  to pause the crawler by calling {@link AutoscaledPool#pause}
- *  or to abort it by calling {@link AutoscaledPool#abort}.
- *
- */
-class PuppeteerCrawler {
-    /**
-     * @param {PuppeteerCrawlerOptions} options
-     * All `PuppeteerCrawler` parameters are passed via an options object.
-     */
-    constructor(options) {
-        ow(options, ow.object.exactShape({
-            handlePageFunction: ow.function,
-            gotoFunction: ow.optional.function,
-            handlePageTimeoutSecs: ow.optional.number,
-            gotoTimeoutSecs: ow.optional.number,
-
-            // AutoscaledPool shorthands
-            maxConcurrency: ow.optional.number,
-            minConcurrency: ow.optional.number,
-
-            // BasicCrawler options
-            requestList: ow.optional.object.validate(validators.requestList),
-            requestQueue: ow.optional.object.validate(validators.requestQueue),
-            maxRequestRetries: ow.optional.number,
-            maxRequestsPerCrawl: ow.optional.number,
-            handleFailedRequestFunction: ow.optional.function,
-            autoscaledPoolOptions: ow.optional.object,
-
-            // PuppeteerPool options and shorthands
-            puppeteerPoolOptions: ow.optional.object,
-            launchPuppeteerFunction: ow.optional.function,
-            launchPuppeteerOptions: ow.optional.object,
-
-            sessionPoolOptions: ow.optional.object,
-            persistCookiesPerSession: ow.optional.boolean,
-            useSessionPool: ow.optional.boolean,
-            proxyConfiguration: ow.optional.object.validate(validators.proxyConfiguration),
-        }));
+class PuppeteerCrawler extends BrowserCrawler {
+    constructor(options = {}) {
+        // @TODO transform options;
+        // @TODO: Should we preserve the options or can we use the PuppeteerCrawler BrowserPool once?
+        // @TODO: Can we throw away the launchFunction?
```

Comment:
We should make the opts user-friendly. Let's talk about the possible structures.